### PR TITLE
Fix PCT Test

### PR DESCRIPTION
**Pull request by mtughan**
HtmlUnit does not appear to support the `[key]` syntax for setting property IDs in ECMAScript objects. To work around this, create an object specifically for the request headers and set the property using the standard syntax instead of setting it during object creation.

Fixes an issue introduced by #491 and will allow jenkinsci/bom#3524 to be merged.

### Testing done

Tested locally using the PCT on the weekly line. Was failing before this change and succeeds afterwards.

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/webapp/js/build.js [hunk 1]
```diff
@@ -7,11 +7,11 @@ if (rebuildButton) {
     if (rebuildButton.dataset.parameterized === 'true') {
       window.location.href = buildUrl
     } else {
+      const headers = {}
```
Comment (timja) [inline src/main/webapp/js/build.js:10]: Ah there’s a JavaScript API for this search fetch in Jenkins core, it can create the header for

Comment (mtughan) [reply src/main/webapp/js/build.js:10]: Know roughly what it is off the top of your head? I'll take a look for it too.

Comment (mtughan) [reply src/main/webapp/js/build.js:10]: Never mind, I think I found it. Will try using it.

**Review by basil — APPROVED** [review]
Reproduced as described in https://github.com/jenkinsci/pipeline-graph-view-plugin/pull/491#issuecomment-2332112999 before this PR. Can no longer reproduce after this PR.

**Comment by mtughan** [comment]
@basil and @timja: not specifically related to this fix, but Linux-based executors seem to be stuck on https://ci.jenkins.io/. Can someone investigate that?

**Pull request merged**